opba-protocols/protocol-bpmn-shared/src/main/java/de/adorsys/opba/protocol/bpmnshared/service/context/ContextUtil.java

```diff
@@ -55,6 +59,32 @@
         return parser.parseExpression(expression, new TemplateParserContext()).getValue(parseContext, resultClass);
     }
 
+    public URI buildAndExpandQueryParameters(String urlTemplate, BaseContext context, String redirectCode, String scaType) {
+        URI uri;
+        if (urlTemplate.contains("{wrong}")) {
+            urlTemplate = urlTemplate.replace("{wrong}", context.getWrongAuthCredentials().toString());
+        }
+        if (urlTemplate.contains("{userSelectScaType}") && scaType != null) {
+            urlTemplate = urlTemplate.replace("{userSelectScaType}", scaType);
```

Comment:
urlTemplate.replace("{userSelectScaType}", scaType); - this is the job of `UriComponentsBuilder`